### ISSUE TITLE
MXE Tutorial

### ISSUE THREAD

**Issue by dgeneratioN** [sm64pc/sm64pc#259]
I just wrote a little tutorial on how to compile with MXE.
While it is working, it requires the user to compile the MXE tools, 
there is a debian repository with the precompiled packages but I 
couldn't get them to work, so these steps are commented out.
[Compiling-on-MXE.md.txt](https://github.com/sm64pc/sm64pc/files/4672833/Compiling-on-MXE.md.txt)

**Comment by coltongit** [comment]
Ah. The precompiled packages use a different CROSS variable which I forgot to account for. That should be an easy fix.

**Comment by dgeneratioN** [comment]
thanks for that! I've learned a lot messing with MXE and the project itself.

**Comment by coltongit** [comment]
You're welcome. I had the same trouble as you with getting a build in my native environment (WSL), so I took to attempting to get the game to build, and after some makefile tweaks (and MXE), I got it to compile and boot for the first time.
I have a PR out that extends MXE support to the precompiled packages and fixes every last compiler warning I could reasonably fix. #252

**Comment by dgeneratioN** [comment]
for some reason mxe-... isn't pointing to the mxe binaries. On my ubuntu 18 WSL they got installed to the following folder
```
pkg-query -L mxe-i686-w64-mingw32.shared-gcc | head | grep gcc
/usr/lib/mxe/usr/bin/i686-w64-mingw32.shared-gcc
```
so maybe the correct step is to add /usr/lib/mxe/usr/bin to the path?

**Comment by coltongit** [comment]
SM64PC uses the static library version:
`mxe-i686-w64-mingw32.static-`
`i686-w64-mingw32.static-`
etc.

**Comment by dgeneratioN** [comment]
yes but even the statics install to that folder, being "mxe-..." only the package names.

**Comment by coltongit** [comment]
Perhaps symlinking the binaries could work

**Comment by basvbeek** [comment]
I made a dockerfile based on [dockcross ](https://github.com/dockcross/dockcross), to easily cross compile for Windows on Linux/WSL. Might come in handy.

Dockerfile.cross:
```Dockerfile
FROM dockcross/windows-static-x64

WORKDIR /usr/src/mxe
RUN make MXE_TARGETS='x86_64-w64-mingw32.static' -j4 cc sdl2 glew

RUN apt-get update && \
    apt-get install -y \
    build-essential \
    git \
    python3 \
    libaudiofile-dev \
    libglew-dev \
    libsdl2-dev \
    bsdmainutils

WORKDIR /work
ENTRYPOINT ["make", "-j4", "WINDOWS_BUILD=1", "CROSS=x86_64-w64-mingw32.static-"]
CMD ["VERSION=us"]

# Build image:
# docker build -t sm64pc -f Dockerfile.cross .
#
# Build sm64pc:
# git clone https://github.com/sm64pc/sm64pc.git
# cd sm64pc
# git checkout nightly
# echo -- place baserom here --
# docker run --rm --mount type=bind,source="$(pwd)",destination=/work --user $UID:$UID sm64pc
```

**Comment by coltongit** [comment]
Ooooh. Looks good actually. That'll make MXE compilation like 10 times easier.

**Comment by AryToNeX** [comment]
```
/usr/lib/mxe/usr/bin/i686-w64-mingw32.static-ld: cannot find -lglew32
/usr/lib/mxe/usr/bin/i686-w64-mingw32.static-ld: cannot find -lOle32
```
Using the guide to crosscompile a 32-bit Windows exe, eventually fixed the PATH to include the correct cross executables.. any idea for this one?

**Comment by dgeneratioN** [comment]
make sure you have glew32 installed

**Comment by coltongit** [comment]
install the MXE glew package for the first and idk about the second

**Comment by AryToNeX** [comment]
now only the second one appears... idk about that one too

**Comment by dgeneratioN** [comment]
try deleting the build folder and trying again

**Comment by AryToNeX** [comment]
nothing... it still errors out there

**Comment by fgsfdsfgs** [comment]
Might a case of case sensitive bullshit. Try changing `-lOle32` to `-lole32` in the Makefile.

**Comment by rawr51919** [comment]
I figured out the cause of the original issue why prepackaged MXE doesn't compile: You need to put this line in your .bashrc or similar:
```
export PATH=/usr/lib/mxe/usr/bin:$PATH
```
@dgeneratioN I think you'll be happy I finally figured that out!
Updated your tutorial to include that step too by the way
[Compiling-on-MXE.md.txt](https://github.com/sm64pc/sm64ex/files/6098498/Compiling-on-MXE.md.txt)

**Comment by Raidenkyu** [comment]
> I made a dockerfile based on [dockcross ](https://github.com/dockcross/dockcross), to easily cross compile for Windows on Linux/WSL. Might come in handy.
> 
> Dockerfile.cross:
> 
> ```dockerfile
> FROM dockcross/windows-static-x64
> 
> WORKDIR /usr/src/mxe
> RUN make MXE_TARGETS='x86_64-w64-mingw32.static' -j4 cc sdl2 glew
> 
> RUN apt-get update && \
>     apt-get install -y \
>     build-essential \
>     git \
>     python3 \
>     libaudiofile-dev \
>     libglew-dev \
>     libsdl2-dev \
>     bsdmainutils
> 
> WORKDIR /work
> ENTRYPOINT ["make", "-j4", "WINDOWS_BUILD=1", "CROSS=x86_64-w64-mingw32.static-"]
> CMD ["VERSION=us"]
> 
> # Build image:
> # docker build -t sm64pc -f Dockerfile.cross .
> #
> # Build sm64pc:
> # git clone https://github.com/sm64pc/sm64pc.git
> # cd sm64pc
> # git checkout nightly
> # echo -- place baserom here --
> # docker run --rm --mount type=bind,source="$(pwd)",destination=/work --user $UID:$UID sm64pc
> ```

I tried this Dockerfile to cross-compile [djoslin0/sm64ex-coop](https://github.com/djoslin0/sm64ex-coop), but got the following error:
```bash
tools/n64graphics: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by tools/n64graphics)
make: *** [Makefile:937: build/us_pc/actors/blue_coin_switch/blue_coin_switch_side.rgba16] Error 1
make: *** Waiting for unfinished jobs....
```

@basvbeek Any ideas?
